synapse/handlers/room.py

```diff
@@ -33,6 +35,19 @@
 
 class RoomCreationHandler(BaseHandler):
 
+    PRESETS_DICT = {
+        RoomCreationPreset.PrivateChat: {
+            "join_rules": JoinRules.INVITE,
+            "history_visibility": "invited",
+            "everyone_ops": False,
+        },
+        RoomCreationPreset.PublicChat: {
+            "join_rules": JoinRules.PUBLIC,
+            "history_visibility": "shared",
+            "everyone_ops": False,
```

Comment:
Would it be nicer to have this as a `namedtuple` rather than an arbitrary `dict`? This would ensure than any new presets we add MUST specify all fields, rather than risking people missing one off and have it crash at runtime with `KeyErrors` due to direct `config[key]` accesses.